framework/src/constraints/EqualValueEmbededConstraint.C

```diff
@@ -0,0 +1,443 @@
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "EqualValueEmbededConstraint.h"
+#include "FEProblem.h"
+#include "DisplacedProblem.h"
+#include "AuxiliarySystem.h"
+#include "SystemBase.h"
+#include "Assembly.h"
+#include "MooseMesh.h"
+#include "Executioner.h"
+#include "AddVariableAction.h"
+
+#include "libmesh/string_to_enum.h"
+#include "libmesh/sparse_matrix.h"
+
+registerMooseObject("MooseApp", EqualValueEmbededConstraint);
+
+template <>
+InputParameters
+validParams<EqualValueEmbededConstraint>()
+{
+  MooseEnum orders(AddVariableAction::getNonlinearVariableOrders());
+  InputParameters params = validParams<NodeElemConstraint>();
+  params.addClassDescription("This is a constraint enforcing overlapping portions of two blocks to "
+                             "have the same variable value");
+  params.set<bool>("use_displaced_mesh") = false;
+  params.addParam<std::string>("model", "glued", "The contact model to use");
+  params.addParam<std::string>("formulation", "default", "The contact formulation");
+  params.addParam<Real>(
+      "penalty",
+      1e8,
+      "The penalty to apply.  This can vary depending on the stiffness of your materials");
+  params.addParam<MooseEnum>("order", orders, "The finite element order");
+
+  return params;
+}
+
+EqualValueEmbededConstraint::EqualValueEmbededConstraint(const InputParameters & parameters)
+  : NodeElemConstraint(parameters),
+    _displaced_problem(parameters.get<FEProblemBase *>("_fe_problem_base")->getDisplacedProblem()),
+    _fe_problem(*parameters.get<FEProblem *>("_fe_problem")),
+    _model(contactModel(getParam<std::string>("model"))),
+    _formulation(contactFormulation(getParam<std::string>("formulation"))),
+    _penalty(getParam<Real>("penalty")),
+    _residual_copy(_sys.residualGhosted()),
+    _mesh_dimension(_mesh.dimension())
+{
+  _overwrite_slave_residual = false;
+}
+
+void
+EqualValueEmbededConstraint::timestepSetup()
+{
+}
+
+void
+EqualValueEmbededConstraint::jacobianSetup()
+{
+}
+
+bool
+EqualValueEmbededConstraint::shouldApply()
+{
+  if (_current_elem != NULL)
+  {
+    // This computes the contact force once per constraint per variable
+    computeContactForce();
+    return true;
+  }
+  return false;
+}
+
+void
+EqualValueEmbededConstraint::computeContactForce()
+{
+  const Node * node = _current_node;
+  unsigned int sys_num = _sys.number();
+  dof_id_type dof_number = node->dof_number(sys_num, _var.number(), 0);
+
+  switch (_model)
+  {
+    case CM_GLUED:
+      switch (_formulation)
+      {
+        case CF_KINEMATIC:
+          _contact_force = -_residual_copy(dof_number);
+          break;
+
+        case CF_PENALTY:
+          _contact_force = _penalty * (_u_slave[0] - _u_master[0]);
+          break;
+
+        default:
+          mooseError("Invalid contact formulation");
```

Comment:
contact -> constraint

Reply:
Replace all usage of 'contact' with something else (constraint?) since this isn't contact.